Adding an auto-update IsAvailable column in movies grid (PR by Alberto Pérez)

AlbertoPV2005 opened the pull request:
- Added new column with checkmarks showing movie availability status
- Column gets data from RentalService when loading
- Now automatically updates when:
  * User rents a movie (checkmark disappears)
  * User returns a movie (checkmark reappears)
- Connected MovieUserControl to refresh grid after rental actions
- Keeps the display synced with actual availability"

AlbertoPV2005 committed:
Add and auto-update IsAvailable column in movies grid

- Added new column with checkmarks showing movie availability status
- Column gets data from RentalService when loading
- Now automatically updates when:
  * User rents a movie (checkmark disappears)
  * User returns a movie (checkmark reappears)
- Connected MovieUserControl to refresh grid after rental actions
- Keeps the display synced with actual availability"

nach8D-edu commented on HobbyManiaManager/Forms/Movie/CatalogForm.cs:
```diff
@@ -48,5 +50,18 @@ private void dataGridViewMoviesList_SelectionChanged(object sender, EventArgs e)
+            ConfigureMoviesDatagrid();
+        }
+
+        public void SetRefreshAction(Action refreshAction)
```
Do we really need this method? We can call directly to SetRefreshAction in MovieUserContro.

AlbertoPV2005 replied:
I'm not sure, Visual itself added that method for me, although what you say makes more sense, today I'll try to fix it.

nach8D-edu commented on HobbyManiaManager/Forms/Movie/MovieUserControl.cs:
```diff
@@ -13,6 +13,7 @@ public partial class MovieUserControl : UserControl
         private Movie Movie;
         private RentalService _service;
         private CustomersRepository _customersRepository;
+        private Action _refreshDataGridView;
```
I really like your aproach using Action.

I would change the name of the property by something more general like _refreshAction.

Imagine we add something else in the action and not only the datagrid is refreshed.

AlbertoPV2005 replied:
Okay, I'll keep that in mind.

nach8D-edu commented on HobbyManiaManager/Forms/Movie/CatalogForm.cs:
```diff
@@ -48,5 +50,18 @@ private void dataGridViewMoviesList_SelectionChanged(object sender, EventArgs e)
+            dataGridViewMoviesList.DataSource = _moviesRepository.GetAll()
+                .Select(m => new MovieDataGridViewModel(m))
+                .ToList();
+            ConfigureMoviesDatagrid();
```
Do we need to execute this in the refresh? Not sure. I'm talking about the Configure...()

AlbertoPV2005 replied:
You're right, it's unnecessary, in the new commit that's already solved.

AlbertoPV2005 committed:
Refactor refresh logic and naming for clarity

- Removed SetRefreshAction method (both in CatalogForm and MovieUserControl)
- Now CatalogForm calls directly at the action in MovieUserControl
- Renamed _refreshDataGridView to _refreshAction for better semantics
- Renamed RefreshDataGridView to RefreshMoviesGrid for clearer intent
- Removed redundant ConfigureMoviesDatagrid call during refreshes
- Now only executes configuration once during initial load

AlbertoPV2005 commented:
I've made a new commit with everything you asked for, simplifying the code and removing unnecessary things. I'd be happy to make any improvements you could think of.

nach8D-edu reviewed (approved):
Looks good to me! 

Assuming everything is tested and working correctly, I will approve and merge the PR
